src/shared/routing/components/navbar/components/navbar-menu/components/icon/icon.component.tsx

```diff
@@ -0,0 +1,39 @@
+import { Box, IconButton, type SvgIconOwnProps } from '@mui/material';
+import { forwardRef, useMemo, type MouseEventHandler } from 'react';
+
+import { ICON_MAPPER, type IconName } from './utils';
+
+interface IIconProps extends Omit<SvgIconOwnProps, 'children'> {
+  name: IconName;
+  disabled?: boolean;
+  padding?: number;
+  height?: number;
+  onClick?: MouseEventHandler<HTMLElement>;
+}
+
+const Icon = forwardRef<HTMLDivElement, IIconProps>(
+  ({ name, disabled, padding, height, onClick, sx, ...rest }, ref) => {
+    const Element = useMemo(() => ICON_MAPPER[name], [name]);
+    const icon = <Element {...rest} sx={{ display: 'block', ...sx }} />;
+
+    return onClick ? (
+      <IconButton
+        onClick={onClick}
+        sx={{ padding, height }}
```

Comment:
I would like to avoid using sx prop to add styles to the component. We can use styled component from mui instead. What do you think @ff-marcela-conejo  @ff-crodriguez